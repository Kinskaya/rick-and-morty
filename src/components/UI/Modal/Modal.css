.modal {
    position: fixed;
    top: 0;
    bottom: 0;
    right: 0;
    left: 0;
    display: none;
    background: rgba(0,0,0,0.5);
}

.modal.active {
    display: flex;
    justify-content: center;
    align-items: center;
}

.modal-content {
    display: contents;
}

.modal-btn {
    position: absolute;
    background-color: transparent;
    margin-left: 225px;
    margin-bottom: 280px;
    font-size: medium;
    border: none;
    cursor: pointer;
}
