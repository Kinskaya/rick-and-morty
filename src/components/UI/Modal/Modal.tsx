import React from 'react';
import classNames from 'classnames';
import './Modal.css';

type TModal = {
  visible: boolean;
};

class Modal extends React.Component<TModal> {
  render() {
    return (
      <div className={classNames('modal', { ['active']: this.props.visible })}>
        <div className="modal-content" onClick={(event) => event.stopPropagation()}>
          {this.props.children}
          <button className="modal-btn">x</button>
        </div>
      </div>
    );
  }
}

export default Modal;
