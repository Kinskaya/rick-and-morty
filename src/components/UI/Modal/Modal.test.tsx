import { render, fireEvent } from '@testing-library/react';
import Modal from './Modal';

describe('Modal component', () => {
  it('render correctly', () => {
    const modal = {
      visible: true,
      handleVisible(): void {},
    };
    const { asFragment } = render(
      <Modal visible={modal.visible} handleVisible={modal.handleVisible} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('modal shows the children and a close button', () => {
    const modal = {
      visible: false,
      handleVisible(): void {},
    };
    const handleClose = jest.fn();

    const { getByText } = render(
      <Modal handleVisible={modal.handleVisible} visible={modal.visible}>
        <div>test</div>
      </Modal>
    );

    expect(getByText('test')).toBeTruthy();

    fireEvent.click(getByText(/x/));

    expect(handleClose).toHaveBeenCalledTimes(0);
  });
});
