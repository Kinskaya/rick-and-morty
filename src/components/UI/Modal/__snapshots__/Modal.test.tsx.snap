// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component render correctly 1`] = `
<DocumentFragment>
  <div
    class="modal active"
  >
    <div
      class="modal-content"
    >
      <button
        class="modal-btn"
      >
        x
      </button>
    </div>
  </div>
</DocumentFragment>
`;
